### ISSUE TITLE
After removing and adding back the universe data messages are not sent

### ISSUE THREAD

**Issue by WojciechTrzasko** [dsmurfin/sACNKit#32]
Hi All 👋 
Nice to meet you ;)

I have a project using sACNKit and found edge case in which the socket termination message is always sent instead of message for universe.

In my case the problem occurs when doing such operations:
1. Start `sACNSource` and add a single universe to it
2. Call `removeUniverse` with the number of above universe
3. Add again the same universe again

As far as I figured out the `removeUniverse` registers socket in the `socketsShouldTerminate` dictionary, which later is never cleaned out. In effect the `sendDataMessages` method instead of picking up the right message, always sends the `socketTerminationMessages`. This happens in this line:
```
let messages = socketsShouldTerminate[interface] != nil ? socketTerminationMessages : universeMessages
```

The intuition says me that probably we should clean the `socketsShouldTerminate` dictionary at some point, but I'm not that familiar with the whole code to be able to quickly find proper place.

If you could take a look I would be more than grateful 🙇 . Or you can leave me a hint where it should be cleaned and I will prepare a PR with the changes. Just let me know what suits you better ;)

**Comment by dsmurfin** [comment]
Hi @WojciechTrzasko nice to meet you too. Thanks for reporting this.

Off the top of my head I think you might be onto the right track with emptying the dictionary. I will take a look at this and look to get this resolved. Universe termination in general is a bit of a tricky problem to resolve properly which is why many implementations don't do it right and just stop sending!

Appreciate you sending this over!

**Comment by WojciechTrzasko** [comment]
Thanks @dsmurfin 🙇 
I was guessing this is not an easy task and this is why I first decided to ask you for help. If you see a place where I could help somehow, just let me know 😉.

**Comment by dsmurfin** [comment]
@WojciechTrzasko this has been resolved in v1.1.1.

I have revised how universe termination is handled. Your issue would have only occured if you had a single present and then removed that, otherwise the dictionary would have been emptied. However, when looking into this I found that non-terminating universes were also being impacted when when terminating another, so I have reworked how this functions. It's now much cleaner.

Thanks again for finding this and letting me know!
